x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -0,0 +1,236 @@
+    });
+
+    const alertTypeRegistry = new AlertTypeRegistry({
+      taskManager: plugins.taskManager,
+      taskRunnerFactory: this.taskRunnerFactory,
+    });
+    this.alertTypeRegistry = alertTypeRegistry;
+    this.serverBasePath = core.http.basePath.serverBasePath;
+
+    core.http.registerRouteHandlerContext('alerting', this.createRouteHandlerContext());
+
+    // Routes
+    const router = core.http.createRouter();
+    // Register routes
+    createAlertRoute(router, this.licenseState);
+    deleteAlertRoute(router, this.licenseState);
+    findAlertRoute(router, this.licenseState);
+    getAlertRoute(router, this.licenseState);
+    getAlertStateRoute(router, this.licenseState);
+    listAlertTypesRoute(router, this.licenseState);
+    updateAlertRoute(router, this.licenseState);
+    enableAlertRoute(router, this.licenseState);
+    disableAlertRoute(router, this.licenseState);
+    updateApiKeyRoute(router, this.licenseState);
+    muteAllAlertRoute(router, this.licenseState);
+    unmuteAllAlertRoute(router, this.licenseState);
+    muteAlertInstanceRoute(router, this.licenseState);
+    unmuteAlertInstanceRoute(router, this.licenseState);
+
+    return {
+      registerType: alertTypeRegistry.register.bind(alertTypeRegistry),
+    };
+  }
+
+  public start(core: CoreStart, plugins: AlertingPluginsStart): PluginStartContract {
+    const {
+      spaces,
+      isESOUsingEphemeralEncryptionKey,
+      logger,
+      taskRunnerFactory,
+      alertTypeRegistry,
+      alertsClientFactory,
+      security,
+    } = this;
+
+    alertsClientFactory.initialize({
+      alertTypeRegistry: alertTypeRegistry!,
+      logger,
+      taskManager: plugins.taskManager,
+      securityPluginSetup: security,
+      encryptedSavedObjectsPlugin: plugins.encryptedSavedObjects,
+      spaceIdToNamespace: this.spaceIdToNamespace,
+      getSpaceId(request: KibanaRequest) {
+        return spaces?.getSpaceId(request);
+      },
+    });
+
+    taskRunnerFactory.initialize({
+      logger,
+      getServices: this.getServicesFactory(core.savedObjects),
+      spaceIdToNamespace: this.spaceIdToNamespace,
+      executeAction: plugins.actions.execute,
+      encryptedSavedObjectsPlugin: plugins.encryptedSavedObjects,
+      getBasePath: this.getBasePath,
+    });
+
+    return {
+      listTypes: alertTypeRegistry!.list.bind(this.alertTypeRegistry!),
+      // Ability to get an alerts client from legacy code
+      getAlertsClientWithRequest(request: KibanaRequest) {
+        if (isESOUsingEphemeralEncryptionKey === true) {
+          throw new Error(
+            `Unable to create alerts client due to the Encrypted Saved Objects plugin using an ephemeral encryption key. Please set xpack.encryptedSavedObjects.encryptionKey in kibana.yml`
+          );
+        }
+        return alertsClientFactory!.create(request, core.savedObjects.getScopedClient(request));
+      },
+    };
+  }
+
+  private createRouteHandlerContext = (): IContextProvider<
```

Comment:
cool - hadn't seen a real use of this yet.  I wonder if we could move more stuff in here over time, like the license check, but that's for another PR :-)